dmutils/filters.py

```diff
@@ -3,3 +3,13 @@
 
 def markdown_filter(text, *args, **kwargs):
     return markdown(text, ['markdown.extensions.abbr'], *args, **kwargs)
+
+
+def smart_join_filter(list_to_join):
+    lower_case = list(map(lambda text: text.lower(), list_to_join))
```

Comment:
I suggest we remove lowercasing from this. There's already a way to lowercase a list of strings using Jinja
(` | map('lower')`) that can be chained with other filters, and only doing join would make this filter more general.